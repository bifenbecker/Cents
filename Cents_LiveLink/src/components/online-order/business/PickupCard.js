import React from "react";
import {Flex, Text, Image, Card, Button} from "rebass/styled-components";
import PropTypes from "prop-types";

import {BlueCheckedIcon, BlueVan} from "../../../assets/images/index";

const PickupCard = props => {
  const {onScheduleClick} = props;

  return (
    <Card {...styles.cardContainer}>
      <Flex {...styles.contentContainer}>
        <Flex {...styles.textContainer}>
          <Text>
            <Image src={BlueCheckedIcon} {...styles.completeIcon}></Image> We can pickup
            from this address!
          </Text>
        </Flex>
        <Image src={BlueVan} {...styles.imageStyles}></Image>
      </Flex>

      <Flex {...styles.footer.wrapper}>
        <Button variant="primary" {...styles.footer.button} onClick={onScheduleClick}>
          Schedule Pickup
        </Button>
      </Flex>
    </Card>
  );
};

const styles = {
  cardContainer: {
    sx: {
      width: "100%",
      p: "18px",
      borderRadius: "24px",
      // backgroundColor: "#FAFAFA",
      boxShadow:
        "0 4px 5px 0 rgba(0,0,0,0.14), 0 1px 10px 0 rgba(0,0,0,0.12), 0 2px 4px -1px rgba(0,0,0,0.2)",
      mt: "18px",
    },
  },
  textContainer: {
    flex: "1",
    fontSize: ["16px", "18px"],
  },
  completeIcon: {
    width: ["16px", "18px"],
    pr: "4px",
  },
  imageStyles: {
    width: "120px",
  },
  contentContainer: {
    alignItems: "flex-start",
    justifyContent: "space-between",
  },
  footer: {
    wrapper: {
      width: "100%",
      mt: "auto",
      alignItems: "center",
      justifyContent: "center",
      pt: "18px",
    },
    button: {
      width: "100%",
      height: "48px",
      fontSize: "16px",
      sx: {
        textTransform: "uppercase",
      },
    },
  },
};

PickupCard.propTypes = {
  onScheduleClick: PropTypes.func.isRequired,
};

export default PickupCard;
