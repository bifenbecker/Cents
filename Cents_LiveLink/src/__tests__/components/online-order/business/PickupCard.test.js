import React from "react";
import {render, fireEvent} from "@testing-library/react";
import PickupCard from "../../../../components/online-order/business/PickupCard";

const renderPickupCardComponent = props => {
  const {getByText} = render(<PickupCard {...props} />);

  return {getByText};
};

describe("Testing PickupCard Component", () => {
  const onScheduleClick = jest.fn();
  it("Test pickupcard UI snapshot", async () => {
    expect(render(<PickupCard onScheduleClick={jest.fn()} />)).toMatchSnapshot();
  });

  it("Schedule Pickup btn is clicked", async () => {
    const {getByText} = renderPickupCardComponent({
      onScheduleClick: onScheduleClick,
    });
    const button = await getByText("Schedule Pickup");
    fireEvent.click(button);

    expect(onScheduleClick).toBeCalled();
  });
});
