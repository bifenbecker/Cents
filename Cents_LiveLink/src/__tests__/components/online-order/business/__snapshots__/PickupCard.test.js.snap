// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PickupCard Component Test pickupcard UI snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ fZSXOh"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi dnioTF hzPtvw"
        >
          <div
            class="sc-bczRLJ sc-gsnTZi wLkkK hzPtvw"
            font-size="16px,18px"
          >
            <div
              class="sc-bczRLJ jITePI"
            >
              <img
                class="sc-bczRLJ dBPgxL"
                src="Icon_Delivery_Available.svg"
                width="16px,18px"
              />
               We can pickup from this address!
            </div>
          </div>
          <img
            class="sc-bczRLJ cRGuKH"
            src="Blue_Van.svg"
            width="120px"
          />
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi jIxXay hzPtvw"
          width="100%"
        >
          <button
            class="sc-bczRLJ iElVFP"
            font-size="16px"
            height="48px"
            width="100%"
          >
            Schedule Pickup
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ fZSXOh"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi dnioTF hzPtvw"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi wLkkK hzPtvw"
          font-size="16px,18px"
        >
          <div
            class="sc-bczRLJ jITePI"
          >
            <img
              class="sc-bczRLJ dBPgxL"
              src="Icon_Delivery_Available.svg"
              width="16px,18px"
            />
             We can pickup from this address!
          </div>
        </div>
        <img
          class="sc-bczRLJ cRGuKH"
          src="Blue_Van.svg"
          width="120px"
        />
      </div>
      <div
        class="sc-bczRLJ sc-gsnTZi jIxXay hzPtvw"
        width="100%"
      >
        <button
          class="sc-bczRLJ iElVFP"
          font-size="16px"
          height="48px"
          width="100%"
        >
          Schedule Pickup
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
